nearcore/src/runtime/mod.rs

```diff
@@ -689,6 +690,30 @@ impl RuntimeAdapter for NightshadeRuntime {
 
         let runtime_config = self.runtime_config_store.get_config(current_protocol_version);
 
+        // To avoid limiting the throughput of the network, we want to include enough receipts to
+        // saturate the capacity of the chunk even in case when all of these receipts end up using
+        // the smallest possible amount of gas, which is at least the cost of execution of action
+        // receipt.
+        // Currently, the min execution cost is ~100 GGas and the chunk capacity is 1 PGas, giving
+        // a bound of at most 10000 receipts processed in a chunk.
+        let delayed_receipts_indices: DelayedReceiptIndices =
+            near_store::get(&state_update, &TrieKey::DelayedReceiptIndices)?.unwrap_or_default();
+        let min_fee = runtime_config.fees.fee(ActionCosts::new_action_receipt).exec_fee();
```

Comment:
This is the correct fee to look at, it's what an empty receipt would cost to execute. (Today, one can submit receipts with an empty action list. Probably we shouldn't allow it as it's useless for users. The smallest gas consumption of a useful receipt would be that of a transfer, which adds another 115 Ggas, so it would give us a factor of ~2 here.)

Anyway, back to the topic :)
I'm not sure if it's a good pattern to use config parameters in the actual code. We are doing it quite a bit, best example is a few lines below where we use `storage_write_value_byte` to limit the tx sizes. But it means changing those parameter values has sometimes unexpected effect on other parts of the code.

If it wasn't for the existing pattern, I'd say we should either use a hard-coded constant or add a new parameter. Where would we add the parameter? I'd probably put it in config.json. But hard coding it seems fine too.

But because we have precedence, I am also completely okay with this as it is. I'll let you decide with what you want to go.

Reply:
I agree with you that these unexpected dependencies on config parameters make code more complex. However, I think we do have this dependency even if we use a constant/new parameter here (as the chosen value need to take into account gas parameters) - it is just tracked more implicitly and not in the code. In the near future, we plan for this limit to be enforced by the protocol (at least the upper bound), so it can't be a part of the per-node config either.